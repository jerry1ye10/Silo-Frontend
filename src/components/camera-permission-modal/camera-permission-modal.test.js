import React from 'react';
import renderer, {act} from 'react-test-renderer';

import {Linking} from 'react-native';
jest.mock('Linking', () => {
  return {
    openSettings: jest.fn(),
  };
});

import {CameraPermissionModal} from '.';

const setModalVisibility = jest.fn();

describe('<CameraPermissionModal />', () => {
  it('modal visibility 1', () => {
    const component = renderer.create(
      <CameraPermissionModal
        modalVisibility={true}
        setModalVisibility={setModalVisibility}
      />,
    );
    expect(component.toJSON()).toMatchSnapshot();
    const WideModalBase = component.root.findByProps({
      testID: 'wide-modal-base',
    });
    expect(WideModalBase.props.modalVisibility).toEqual(true);
  });

  it('modal visibility 2', () => {
    const component = renderer.create(
      <CameraPermissionModal
        modalVisibility={false}
        setModalVisibility={setModalVisibility}
      />,
    );
    expect(component.toJSON()).toMatchSnapshot();
    const WideModalBase = component.root.findByProps({
      testID: 'wide-modal-base',
    });
    expect(WideModalBase.props.modalVisibility).toEqual(false);
  });

  it('cancel button pressed', () => {
    const component = renderer.create(
      <CameraPermissionModal
        modalVisibility={true}
        setModalVisibility={setModalVisibility}
      />,
    );
    const cancelButton = component.root.findByProps({
      testID: 'cancel-button',
    });
    cancelButton.props.onPress();
    expect(setModalVisibility).toHaveBeenCalledWith(false);
  });

  it('settings button pressed with modal hide', () => {
    const component = renderer.create(
      <CameraPermissionModal
        modalVisibility={true}
        setModalVisibility={setModalVisibility}
      />,
    );
    const settingsButton = component.root.findByProps({
      testID: 'settings-button',
    });
    act(() => {
      settingsButton.props.onPress();
    });
    expect(setModalVisibility).toHaveBeenCalledWith(false);
    const WideModalBase = component.root.findByProps({
      testID: 'wide-modal-base',
    });
    act(() => {
      WideModalBase.props.onModalHide();
    });
    expect(Linking.openSettings).toHaveBeenCalled();
  });
});
