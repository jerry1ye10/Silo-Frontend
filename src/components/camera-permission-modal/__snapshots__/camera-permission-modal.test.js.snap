// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CameraPermissionModal /> modal visibility 1 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  testID="modal"
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    clickable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      Array [
        Array [
          Object {
            "margin": 37.5,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          },
          Object {
            "flex": 1,
            "justifyContent": "center",
          },
          undefined,
          Object {
            "margin": 0,
          },
        ],
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      hideModalContentWhileAnimating={false}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      pointerEvents="box-none"
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "margin": 37.5,
          "transform": Array [
            Object {
              "translateY": 1334,
            },
          ],
        }
      }
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
      testID="modal"
    >
      <View
        onLayout={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 20,
          }
        }
        testID="container"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#f8f8f8",
              "borderRadius": 2,
              "height": 250,
              "overflow": "hidden",
              "paddingHorizontal": 10,
              "paddingVertical": 20,
              "width": 300,
            }
          }
          testID="modal-view"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "space-between",
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                }
              }
            >
              Enable Camera
            </Text>
            <Text
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              Labyrinthe needs access to your
              

              camera to unlock desks in store.
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "width": "100%",
                }
              }
            >
              <View
                accessible={true}
                clickable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "backgroundColor": "green",
                    "borderRadius": 2,
                    "margin": 5,
                    "opacity": 1,
                    "paddingVertical": 10,
                    "width": "90%",
                  }
                }
                testID="settings-button"
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontWeight": "bold",
                      "textAlign": "center",
                    }
                  }
                >
                  SETTINGS
                </Text>
              </View>
              <View
                accessible={true}
                clickable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "margin": 5,
                    "opacity": 1,
                    "paddingVertical": 10,
                    "width": "90%",
                  }
                }
                testID="cancel-button"
              >
                <Text
                  style={
                    Object {
                      "color": "red",
                      "fontWeight": "bold",
                      "textAlign": "center",
                    }
                  }
                >
                  CANCEL
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`<CameraPermissionModal /> modal visibility 2 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  testID="modal"
  transparent={true}
  visible={false}
>
  <View
    accessible={true}
    clickable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      Array [
        Array [
          Object {
            "margin": 37.5,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          },
          Object {
            "flex": 1,
            "justifyContent": "center",
          },
          undefined,
          Object {
            "margin": 0,
          },
        ],
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      hideModalContentWhileAnimating={false}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      pointerEvents="box-none"
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "margin": 37.5,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
      testID="modal"
    >
      <View
        onLayout={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 20,
          }
        }
        testID="container"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#f8f8f8",
              "borderRadius": 2,
              "height": 250,
              "overflow": "hidden",
              "paddingHorizontal": 10,
              "paddingVertical": 20,
              "width": 300,
            }
          }
          testID="modal-view"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "space-between",
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                }
              }
            >
              Enable Camera
            </Text>
            <Text
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              Labyrinthe needs access to your
              

              camera to unlock desks in store.
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "width": "100%",
                }
              }
            >
              <View
                accessible={true}
                clickable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "backgroundColor": "green",
                    "borderRadius": 2,
                    "margin": 5,
                    "opacity": 1,
                    "paddingVertical": 10,
                    "width": "90%",
                  }
                }
                testID="settings-button"
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontWeight": "bold",
                      "textAlign": "center",
                    }
                  }
                >
                  SETTINGS
                </Text>
              </View>
              <View
                accessible={true}
                clickable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "margin": 5,
                    "opacity": 1,
                    "paddingVertical": 10,
                    "width": "90%",
                  }
                }
                testID="cancel-button"
              >
                <Text
                  style={
                    Object {
                      "color": "red",
                      "fontWeight": "bold",
                      "textAlign": "center",
                    }
                  }
                >
                  CANCEL
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
